main.js

```diff
@@ -0,0 +1,280 @@
+'use strict'
+var DEBUG = false
+if(DEBUG){
+  var dbg = document.createElement('p')
+  dbg.textContent = 'default text'
+  dbg.style.position = 'absolute'
+  dbg.style.top = '0px'
+  document.querySelector('body').appendChild(dbg)
+}
+
+function dbgt() {
+  var argumentString = ''
+  for(var i of arguments) {
+    argumentString += i
+  }
+  dbg.textContent = '$ '+argumentString
+  console.log(argumentString)
+}
+
+(function(){
+  var originalErrFunction = console.error
+
+  console.error = function() {
+    var debugMessage = ['Error: ']
+    for(var i of arguments) {
+      debugMessage.push(i)
+    }
+    dbgt(debugMessage.join(''))
+    originalErrFunction.apply(null, arguments)
+  }
+}())
+
+function Calculator() {
+  this.operands = []
+  this.tailIndex = 0
+  this.inputSize = 3.0
+  this.clearOrAllClear = 'all'
+  this.domElements = {
+    'Numbers': [
+      document.querySelector('.calculator-row:nth-child(5) > .calculator-field:nth-child(1) > button'),
+      document.querySelector('.calculator-row:nth-child(4) > .calculator-field:nth-child(1) > button'),
```

Comment:
Gotcha. Perhaps I should spend a little extra time reading ahead on Mondays to get an overview of how the project will change, and reduce how much refactoring I have to do.